qemu/tests/balloon_service.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import time
+
+from virttest import utils_test
+from virttest import qemu_monitor
+from virttest import error_context
+from avocado.core import exceptions
+from autotest.client.shared import utils
+from autotest.client.shared import error
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Balloon service test for windows guest.
+    1) boot a windows guest with balloon device.
+    2) enable and check driver verifier in guest.
+    3) install balloon service in guest.
+    4) enable polling in qmp.
+    5) evict and enlarge balloon.
+    6) get polling value in qmp.
+    7) uninstall balloon service and clear driver verifier.
+    """
+
+    def balloon_service(session):
+        """
+        Check / Install balloon service.
+
+        :param session: VM session.
+        """
+        status_balloon_service = params.get("status_balloon_service")
```

Comment:
make  your logical more clearly. I think it should be
  if not running, check is it installed else install it and check is it running